src/server/pegasus_server_impl.cpp

```diff
@@ -2274,7 +2294,8 @@ pegasus_server_impl::append_key_value_for_scan(std::vector<::dsn::apps::key_valu
                                                uint32_t epoch_now,
                                                bool no_value,
                                                bool request_validate_hash,
-                                               bool request_expire_ts)
+                                               bool request_expire_ts,
+                                               bool fill_value)
```

Comment:
The function name `append_key_value_for_scan` is not suitable now, because the 'append' operation will not happen when fill_value is false.
You can do a refactor to separete then key-value valicate part and append part into two functions.